tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,190 @@
+---
+title: Golden testing in Haskell with hspec-golden
+published: 2019-08-10
+tags: haskell, golden testing, snapshot testing, expect testing
+libraries: hspec-golen-0.1.0.0
+language: haskell
+author: Adrian Vera
+author-name: Adrian Vera
+twitter-profile: adrian_med
+github-profile: medAdrian
+description: In this tutorial we will implement golden tests in Haskell using hspec-golden a super set of the known test framework hspec. A golden test is an Input/Output action that records the result of a test into a file known as the "Golden File".
+---
+
+Golden testing (also known as Golden master testing or Snapshot testing) is pretty similar to unit testing, except that the test output is stored in a separate file known as the "Golden File".
+
+This kind of testing will help us verify modifications made to a reference version of our SUT(Subject Under Test) and validate that it has not modified its
+behaviour in an incorrect way. Thus giving us some safety when extending and refactoring code that doesn't have proper tests.
+
+When we write golden tests we build up knowledge of what the code is doing, this is very useful when we want to re-write it,
+we can run our test to find out immediately whether we've modified the desired behaviour (we start to see our tests in a different way
+since the purpose of golden testing is to document your system's behaviour, not to check for the wished one).
+
+Once we adopt this perspective, we have a completely different frame for understanding automated tests in general,
+we can see tests as descriptions of what we have rather than statements of correctness.
+We can review our tests periodically to tighten up their conditions as we decide what the behavior should be at each level of our systems.
+
+##The Main idea
+
+####‘A Golden File is your source of truth. You validate your test response against the Golden File. To summarize, the Golden file contains the response you expect from your test’
+Source: [Medium](https://medium.com/@jarifibrahim/golden-files-why-you-should-use-them-47087ec994bf) 
+
+##Changes on test flow
```

Comment:
@MedAdrian Are this headers displaying correctly in your end? I tried to view the file using the GitHub preview file: https://github.com/stackbuilders/tutorials/blob/5c8539d65fc09682ff11573c14ac67081e8e4ed2/tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md But they look like raw strings: `##The Main idea`

Reply:
I've tested the tutorial.md on vscode and Webstorm, the preview seems right but it seems it's not getting the markdown on github.

![image](https://user-images.githubusercontent.com/13720528/63170073-110a5900-bffe-11e9-8412-25914b7ec633.png)

Even so, i will add a space between the numeral sign and the text to avoid any issues.
### i.e
`###The Main Idea` will become `### The Main Idea`

Reply:
What do you think making use of another type of styling?, such as blockquote for example.

Reply:
I think it looks better the way @po5i proposed, @CristhianMotoche what do you think?

![image](https://user-images.githubusercontent.com/13720528/63174139-fab4cb00-c006-11e9-9126-8550a0b81011.png)